pkg/apis/kibana/v1/kibana_types.go

```diff
@@ -73,7 +72,19 @@ func (k Kibana) IsMarkedForDeletion() bool {
 	return !k.DeletionTimestamp.IsZero()
 }
 
-func (k *Kibana) ElasticsearchRef() commonv1.ObjectSelector {
+func (k *Kibana) Associated() commonv1.Associated {
+	return k
+}
+
+func (k *Kibana) AnnotationName() string {
+	return "association.k8s.elastic.co/es-conf"
+}
+
+func (k *Kibana) AssociatedServiceType() string {
+	return "elasticsearch"
+}
+
+func (k *Kibana) AssociationRef() commonv1.ObjectSelector {
```

Comment:
This is not an easy change because it would also introduce some changes in the [Kibana driver](https://github.com/elastic/cloud-on-k8s/blob/2841f5eb9002b21c4c73cc2987741768efa27c56/pkg/controller/kibana/driver.go#L202) and in the [configuration generation](https://github.com/elastic/cloud-on-k8s/blob/master/pkg/controller/kibana/config_settings.go#L90) _(same for EnterpriseSearch I guess)_.

My feeling is that iterating over the APM associations in the APMServer controller makes sense. But since there is only one association here I think it's fair for `Kibana` _(or EnterpriseSearch)_ to implement the `Association` interface. It also helps to keep this PR smaller.